extra_data/sourcedata.py

```diff
@@ -22,14 +22,15 @@ class SourceData:
     _first_source_file = ...
 
     def __init__(
-            self, source, *, sel_keys, train_ids, files, section,
-            is_single_run, inc_suspect_trains=True
+            self, source, *, sel_keys, train_ids, files, section, legacy,
+            is_single_run, inc_suspect_trains=True,
     ):
         self.source = source
         self.sel_keys = sel_keys
         self.train_ids = train_ids
         self.files: List[FileAccess] = files
         self.section = section
+        self.legacy = legacy
```

Comment:
Am I right in thinking that the value of `.legacy` is the new, preferred name whereas `.source` is the legacy name? That seems basically guaranteed to cause confusion. :wink: 

Would it work to skip constructing `SourceData` objects for legacy sources, and when you do `dc[legacy_src]`, you just get back the SourceData with the new name, a bit like another kind of alias?

If not that, let's at least come up with a better name for the attribute.

Reply:
> Would it work to skip constructing `SourceData` objects for legacy sources, and when you do `dc[legacy_src]`, you just get back the SourceData with the new name, a bit like another kind of alias?

👀 
I would actually prefer that approach, but was hesitant to be radical. Happy to have a closer look at the needed changes.

Reply:
I suspect it will be more work in some ways, because it implies you have to carry a dict of legacy names through all of the DataCollection -> DataCollection operations (select, union, with_aliases...). With the present approach you can reconstruct it from the sources after each of those, and there are fewer SourceData -> SourceData operations to fix.

But it might be neater once that's done.

Reply:
As we also discussed in person, I did not go with a more radical approach given the complexity it introduces with `DataCollection.select()`. We'd either have to keep references to de-selected data around without actually using them in most places, or replace selections automatically by their canonical names. The latter however feels even more instrusive than returning a different `SourceData` object, as it would e.g. result in

```python3
mod0_name not in run.select(mod0_name).all_sources
```

Still, I changed the `SourceData.legacy` property to `SourceData.canonical_name`, which is populated for all `SourceData` objects now. `SourceData.is_legacy` is `True` if and only if this differs from `SourceData.source`.